src/nisarqa/utils/utils.py

```diff
@@ -480,6 +481,60 @@ def set_logger_handler(
         log.addHandler(handler)
 
 
+def log_function_start_and_stop_time(func):
+    """
+    Function decorator which logs the start and completion of a function.
+
+    The function's arguments are also logged. Useful for benchmarking; the
+    log file can be parsed for timings.
+    """
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+
+        # Construct the "arguments" string for the log message
+        def trunc(arg) -> str:
+            max_length = 40
+            if isinstance(arg, nisarqa.YamlParamGroup):
+                max_length = 100
+            a = f"{arg!r}"
+            if len(a) < max_length or isinstance(
+                arg, (h5py.Dataset, nisarqa.ComplexFloat16Decoder)
+            ):
+                return arg
+            else:
+                return f"{a[:max_length]}[TRUNCATED]"
```

Comment:
Note that you could avoid the quotation marks around truncated argument strings by using string concatenation instead of f-strings (though the implementation could get a bit messy).